### PR TITLE
Skal regne ut alder av ident i soknad-api i stedet for i frontend

### PR DESCRIPTION
**Pull request by throndi**
favro: https://favro.com/organization/98c34fb974ce445eac854de0/a64c6aad9b0d61ef6c0290bd?card=NAV-11927

### CODE REVIEW

**Comment by ViktorGSolberg** [comment]
Ligger fixen i denne comitten? Jeg ønsker gjerne at du lager en egen PR hvor du kjører ktlint. Deretter oppretter du en ny PR fra den gamle og gjør selve fixen der slik at man får adskilt ktlint og fix

**Comment by throndi** [comment]
> Ligger fixen i denne comitten? Jeg ønsker gjerne at du lager en egen PR hvor du kjører ktlint. Deretter oppretter du en ny PR fra den gamle og gjør selve fixen der slik at man får adskilt ktlint og fix

yep, ktlint endringer: https://github.com/navikt/familie-ef-soknad-api/pull/1000

File: src/main/kotlin/no/nav/familie/ef/søknad/person/mapper/SøkerinfoMapper.kt [hunk 1]
```diff
@@ -175,19 +176,24 @@ internal class SøkerinfoMapper(
         )
     }
 
+    private fun kalkulerAlder(ident: String) = Period.between(Fødselsnummer(ident).fødselsdato, LocalDate.now()).years
```
Comment (ViktorGSolberg) [inline src/main/kotlin/no/nav/familie/ef/søknad/person/mapper/SøkerinfoMapper.kt:179]: Skriv gjerne en test på at alderen blir kalkulert riktig. Prøv gjerne med flere forskjellige fødselsdatoer slik at du får testet mulige edge-caser.

File: src/test/kotlin/no/nav/familie/ef/søknad/person/mapper/SøkerinfoMapperTest.kt [hunk 2]
```diff
@@ -134,17 +135,62 @@ internal class SøkerinfoMapperTest {
...
+        val fødselsdato = LocalDate.of(1990, 1, 1)
+        val fødselsnummer = FnrGenerator.generer(fødselsdato)
+        every { EksternBrukerUtils.hentFnrFraToken() } returns fødselsnummer
+
+        val pdlSøker = opprettPdlSøker()
+        val person = søkerinfoMapper.mapTilSøkerinfo(pdlSøker, emptyMap(), emptyMap()).søker
+        val forventetAlder = Period.between(fødselsdato, LocalDate.now()).years
+
+        assertThat(person.alder).isEqualTo(forventetAlder)
+    }
```
Comment (ViktorGSolberg) [inline src/test/kotlin/no/nav/familie/ef/søknad/person/mapper/SøkerinfoMapperTest.kt:175]: Denne testen sjekker nøyaktig det samme som testen under, så denne kan fjernes

File: src/test/kotlin/no/nav/familie/ef/søknad/person/mapper/SøkerinfoMapperTest.kt [hunk 3]
```diff
@@ -134,17 +135,62 @@ internal class SøkerinfoMapperTest {
...
+        val fødselsdato = LocalDate.of(1990, 1, 1)
+        val fødselsnummer = FnrGenerator.generer(fødselsdato, erDnummer = true)
+        every { EksternBrukerUtils.hentFnrFraToken() } returns fødselsnummer
+
+        val pdlSøker = opprettPdlSøker()
+        val person = søkerinfoMapper.mapTilSøkerinfo(pdlSøker, emptyMap(), emptyMap()).søker
+        val forventetAlder = Period.between(fødselsdato, LocalDate.now()).years
+
+        assertThat(person.alder).isEqualTo(forventetAlder)
+    }
```
Comment (ViktorGSolberg) [inline src/test/kotlin/no/nav/familie/ef/søknad/person/mapper/SøkerinfoMapperTest.kt:162]: Denne testen sjekker nøyaktig det samme som testen over, så denne kan også fjernes

**Review by ViktorGSolberg — APPROVED** [review]
Det ser bra ut, ville slettet de testene jeg markerte 🚀 

Et par tips angående det å skrive tester:

- Generelt så ønsker man å skrive tester som dekker normaltilfellene og tester som dekker grensetilfellene. Det er egentlig ikke vits med flere tester som tester det samme normaltilfellet fordi de egentlig tester det samme :) 

- I det tilfellet her så kunne et grensetilfelle kanskje vært å teste en dato som er nøyaktig, si, 20 år tilbake i tid slik at man får sjekket at alderen blir riktig på dagen - så en slik test oppfordrer jeg til at du tar med

File: src/test/kotlin/no/nav/familie/ef/søknad/person/mapper/SøkerinfoMapperTest.kt (hunk 3 above)
Comment (throndi) [reply src/test/kotlin/no/nav/familie/ef/søknad/person/mapper/SøkerinfoMapperTest.kt:162]: Egentlig ikke, siden det er d-nummer i stedet for fnr

File: src/test/kotlin/no/nav/familie/ef/søknad/person/mapper/SøkerinfoMapperTest.kt (hunk 2 above)
Comment (throndi) [reply src/test/kotlin/no/nav/familie/ef/søknad/person/mapper/SøkerinfoMapperTest.kt:175]: Egentlig ikke, siden det er fnr i stedet for dnr

**Comment by throndi** [comment]
> Det ser bra ut, ville slettet de testene jeg markerte 🚀
> 
> Et par tips angående det å skrive tester:
> 
> * Generelt så ønsker man å skrive tester som dekker normaltilfellene og tester som dekker grensetilfellene. Det er egentlig ikke vits med flere tester som tester det samme normaltilfellet fordi de egentlig tester det samme :)
> * I det tilfellet her så kunne et grensetilfelle kanskje vært å teste en dato som er nøyaktig, si, 20 år tilbake i tid slik at man får sjekket at alderen blir riktig på dagen - så en slik test oppfordrer jeg til at du tar med

Her så testes egentlig ikke de samme normaltilfellene, men at alder kalkuleres på en måte som tar høyde for forskjellige individnummer.  Det testes også at det taes høyde for d-nummer.

File: src/test/kotlin/no/nav/familie/ef/søknad/person/mapper/SøkerinfoMapperTest.kt (hunk 3 above)
Comment (throndi) [reply src/test/kotlin/no/nav/familie/ef/søknad/person/mapper/SøkerinfoMapperTest.kt:149]: men fjernet en tilsvarende test for 2000 tallet
